app/src/main/java/com/sanjeevyadavit/magecart/viewmodel/LoginViewModel.kt

```diff
@@ -0,0 +1,69 @@
+package com.sanjeevyadavit.magecart.viewmodel
+
+import android.app.Application
+import android.util.Log
+import android.widget.Toast
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import com.sanjeevyadavit.magecart.model.LoginBodyRequest
+import com.sanjeevyadavit.magecart.service.ApiInterface
+import com.sanjeevyadavit.magecart.service.RetrofitClient
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import retrofit2.Call
+import retrofit2.Response
+
+class LoginViewModel(app: Application): AndroidViewModel(app) {
+    val email: MutableLiveData<String> = MutableLiveData("")
+    val password: MutableLiveData<String> = MutableLiveData("")
```

Comment:
Doing this makes it vulnarable. activity can directly change this value from inside of it, but viewModel's data should only be changed from inside viewModel's function.
```
private val _myData: MutableLiveData<Boolean> = MutableLiveData()
val myData: LiveData: LiveData<Boolean>
    get() = _myData
```
Doing things like above prevents data from being modified from outside as outside we have exposed data that is not mutable and inside of the viewModel you can change `_myData`

Reply:
But I want this `email` and `password` to attach to xml via databinding, if I make it private how will I use two ways data binding?